src/celeritas/ext/VecgeomParams.cc

```diff
@@ -82,6 +83,40 @@ VecgeomParams::VecgeomParams(const std::string& filename)
         vgdml::Frontend::Load(filename, validate_xml_schema);
     }
 
+    Initialize();
+}
+
+#ifdef CELERITAS_USE_GEANT4
+//---------------------------------------------------------------------------//
+/*!
+ *  Create a VecGeom model from an pre-existing Geant4 geometry
+ */
+VecgeomParams::VecgeomParams(const G4VPhysicalVolume* p_G4world)
+{
+    CELER_LOG(info) << "Creating VecGeom model from pre-existing G4 geometry";
+    CELER_ASSERT(p_G4world);
```

Comment:
Since this is a check on the input:
```suggestion
    CELER_EXPECT(p_G4world);
```
and move to the top of the function